librz/asm/arch/tricore/tricore.inc

```diff
@@ -44,12 +50,32 @@ static inline const char *tricore_get_op_regname(csh h, cs_insn *insn, int idx)
 static inline st32 tricore_get_op_imm(cs_insn *insn, int idx) {
 	cs_tricore_op *op = tricore_get_op(insn, idx);
 	if (op->type != TRICORE_OP_IMM) {
-		rz_warn_if_reached();
+		RZ_LOG_WARN("Failed to get operand%d [%d]: \"%s %s\" [imm]\n",
+			idx, tricore_op_count(insn), insn->mnemonic, insn->op_str);
 		return 0;
 	}
 	return op->imm;
 }
 
-static inline ut8 tricore_op_count(cs_insn *insn) {
-	return insn->detail->tricore.op_count;
+static inline csh tricore_setup_cs_handle(const char *cpu, const char *features) {
+	static csh handle = 0;
```

Comment:
Is it possible not to use `static`? Could be done in a separate PR. See https://github.com/rizinorg/rizin/issues/3721

Reply:
Possibly, but then we'd have to save the csh in RzAsm and RzAnalysis or choose not to cache the csh?

Reply:
Good point about caching. It's probably better to add a new member to the RzAnalysis/RzAsm plugin then. Like we have `il_config` for RzIL information, maybe we can have `arch_config` for different things, including capstone. @wargio @ret2libc @thestr4ng3r what do you think? See also https://github.com/rizinorg/rizin/commit/2c9a7008e9e30a10ff42c0422b1ef7d6579b8dab

Reply:
@imbillow nevermind, see `librz/analysis/p/analysis_arm_cs.c` for the example how to use context to save the handle.